### PR TITLE
[WIP] (Auto)generate a .phar

### PR DESCRIPTION
**Pull request by matthiasnoback**
See #126. I've got this working with some minor adaptations.

I'm using https://github.com/box-project/box2 If you follow the installation instructions over there, you can finally run `box build` and this will generate `churn.phar`, which can be used in the same way as `churn`.

Apparently it can even be incorporated into the release cycle on Travis: https://github.com/FriendsOfPHP/pickle/blob/master/.travis.yml, meaning we can have a fresh `.phar` file generated for each GitHub release. I have no experience with this, but I'll try to figure it out.

### CODE REVIEW

**Comment by matthiasnoback** [comment]
Some more progress :) It's always horrible to figure out Travis configuration. But the .travis.yml file from Travis, combined with https://docs.travis-ci.com/user/deployment/releases/ and some issues in the Travis issue tracker resulted in a working deployment. See release "test6-dev": https://github.com/matthiasnoback/churn-php/releases/tag/test6-dev (check out the churn.phar that was created as part of the "deploy" phase of the Travis build. See also: https://travis-ci.org/matthiasnoback/churn-php/builds/284354997#L875

I'll make sure to wrap this PR up as there's a lot of noise because of all the trial and error.
Also, this could use a hint in the README file to point out the option to download the .phar.

**Comment by bmitch** [comment]
@matthiasnoback Looking good. I am used to using GitLab for my CI so using Travis is a bit frustrating for me.

**Comment by ruudwame** [comment]
@bmitch /@matthiasnoback Is this ready to be merged?
After reading Matthias's recent article ([Keep an eye on the churn; finding legacy code monsters](https://matthiasnoback.nl/2018/01/churn-legacy-code/)) I would love to try this tool on a older Symfony project (2.7), and a standalone phar would be really helpful.

**Comment by bmitch** [comment]
@ruudwame I believe it is still a work in progress by @matthiasnoback but not sure if he is planning on continuing it?

Another possible solution for you to run it standalone is via Docker:
```
docker run --rm -ti -v $PWD:/app dockerizedphp/churn run src
```

Does that work for you?

**Comment by ruudwame** [comment]
As Docker isn't some we currently use in our technology stack I dismissed this options (that's clearly listed in the readme) as 'too complicated'.
I'm happy to report that this assumption was completely wrong :).

Running php-churn in a Docker container is very easy and worked flawlessly. (I only had a, not so ver helpful, exception; _"In Assert.php line 942: Expected a value greater than 0. Got: 0"_, which turned out to be a missing .git directory).

So thanks for the tip!

I still think a phar file would be useful tough, as this might match someones workflow a little bit better.

**Comment by bmitch** [comment]
@ruudwame that's great to hear! Glad that worked.

I'll open an issue for that error you got because of the missing '.git' file.

And I'll leave this open in case @matthiasnoback is still working on it.

**Comment by matthiasnoback** [comment]
Hey everyone, some things have changed since I created this PR so I'd have
to take a closer look. Still think it's useful to have a phar for people,
even though the Docker image is very helpful indeed!

Unrelated: I recently ran the analysis on a project and wondered if it
couldn't run much faster. I suspect that starting all those different
processes may produce a lot of unnecessary overhead.

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,49 @@
...
+# - 7.1
+# - 7.2
+# - nightly
 
 before_script:
   - travis_retry composer self-update
-  - travis_retry composer install --no-interaction --prefer-source --dev
+  - travis_retry composer install --no-interaction --prefer-dist --dev
+  # Install box (https://box-project.github.io/box2/)
+  - composer global require 'kherge/box=~2.4' --prefer-dist
```
Comment (theofidry) [inline .travis.yml]: there is a new (and much faster among other things) version for which you don't have to worry about installing only the non-dev dependencies either: https://github.com/humbug/box :)

**Comment by villfa** [comment]
Thank you for your work @matthiasnoback but I'm closing since the PHAR generation has been implemented with #267.

**Pull request closed without merging**